examples/http-circuit-breaker/http_circuit_breaker.bal

```diff
@@ -6,92 +6,90 @@ import ballerina/runtime;
+            statusCodes: [400, 404, 500]
 
         },
-        // The threshold for request failures.
-        // When this threshold exceeds, the circuit trips.
-        // This is the ratio between failures and total requests
-        //  and the ratio is considered only within the configured
-        // `RollingWindow`
-        failureThreshold: 0.2,
-
-        // The time period(in milliseconds) to wait before
-        // attempting to make another request to the upstream service.
-        // When the failure threshold exceeds, the circuit trips to
-        // OPEN state. Once the circuit is in OPEN state
-        // circuit breaker waits for the time configured in `resetTimeMillis`
-        // and switch the circuit to the HALF_OPEN state.
-        resetTimeMillis: 10000,
-
-        // HTTP response status codes which are considered as failures
-        statusCodes: [400, 404, 500]
-
-    },
-    timeoutMillis: 2000
-};
+        timeoutMillis: 2000
+    });
 
 // Create an HTTP service bound to the endpoint (circuitBreakerEP).
 @http:ServiceConfig {
     basePath: "/cb"
 }
-service<http:Service> circuitbreaker bind { port: 9090 } {
+service circuitbreaker on new http:Listener(9090) {
     // Create a REST resource within the API.
     @http:ResourceConfig {
         methods: ["GET"],
         path: "/"
     }
     // The parameters include a reference to the caller
-    // endpoint and an object of the request data.
-    invokeEndpoint(endpoint caller, http:Request request) {
-        var backendRes = backendClientEP->forward("/hello", request);
-        // `match` is used to handle union-type returns.
+    // and an object of the request data.
+    resource function invokeEndpoint(http:Caller caller, http:Request request) {
+
+        var backendResponse = backendClientEP->forward("/hello", request);
+
+        // `is` operator is used to separate out union-type returns.
         // If a response is returned, the normal process runs. If the service
         // does not get the expected response, the error-handling logic is executed.
-        match backendRes {
+        if (backendResponse is http:Response) {
 
-            http:Response res => {
-                var result = caller->respond(res);
-                if (result is error) {
-                   log:printError("Error sending response", err = result);
-                }
+            var responseToCaller = caller->respond(backendResponse);
+            if (responseToCaller is error) {
+                log:printError("Error sending response", err = responseToCaller);
             }
-            error responseError => {
-                http:Response response = new;
-                response.statusCode = 500;
-                response.setPayload(responseError.message);
-                var result = caller->respond(response);
-                if (result is error) {
-                   log:printError("Error sending response", err = result);
-                }
+        } else {
```

Comment:
Should we use else if (backendResponse is error) clause here? Would be much readable if we clearly show the type there specially in BBEs

Reply:
I think, we shouldn't use else if (backendResponse is error) here as it is not the practice that we follow when writing Ballerina code. What we can do is, explain the behavior in a comment in the top of the if check.